pkg/operator/starter.go

```diff
@@ -21,24 +20,25 @@ import (
 
 	operatorclientinformers "github.com/openshift/cluster-openshift-controller-manager-operator/pkg/generated/informers/externalversions"
 	"github.com/openshift/cluster-openshift-controller-manager-operator/pkg/operator/v311_00_assets"
-	"github.com/openshift/library-go/pkg/operator/status"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
 	"github.com/openshift/library-go/pkg/operator/v1alpha1helpers"
+	status "github.com/openshift/library-go/pkg/operator/v1alpha1status"
 )
 
-func RunOperator(clientConfig *rest.Config, stopCh <-chan struct{}) error {
-	kubeClient, err := kubernetes.NewForConfig(clientConfig)
+func RunOperator(ctx *controllercmd.ControllerContext) error {
+	kubeClient, err := kubernetes.NewForConfig(ctx.KubeConfig)
```

Comment:
New `ControllerContext` gives us the KubeConfig, stop channel, and the recorder.

For future unit tests we can use `events.NewInMemoryRecorder` to fake the recorder, or use it to inspect the actions taken.